crates/server/src/store/in_memory/authority.rs

```diff
@@ -428,6 +428,7 @@ impl InnerInMemory {
             .map(|rrset| {
                 let mut new_answer =
                     RecordSet::new(name.borrow(), rrset.record_type(), rrset.ttl());
+                new_answer.set_ttl(rrset.ttl());
 
```

Comment:
@bluejekyll / @djc I am not an expert but line 430 is it correct to pass `rrset.ttl()` where `serial` is expected?

Also I found out that `rrset` has the good value from `wildcard` here 120 but got forgotten by default `0` it explain my fix here.

Reply:
I'm a little confused right here, it looks like the TTL is being set correctly when the new_answer is constructed?

Reply:
I am confused too! The third parameter of `RecordSet` is the `Serial` but we pass here the `TTL` with `rrset.ttl()`.

https://github.com/bluejekyll/trust-dns/blob/a88a7a575a1e77fa91e47abe4fd744db56110bc8/crates/proto/src/rr/rr_set.rs#L31-L57

This explain my change line 431.

Reply:
I am investigating why ttl is not set here: https://github.com/darnuria/trust-dns/blob/5e38e67e13bccbb15191280924718e8908180d48/crates/server/src/store/in_memory/authority.rs#L449-L453

Reply:
Sorry for the `push -f` just found that a function `RecordSet::with_ttl` existed.

Reply:
Fail in CI due to uneeded import will fix it after a local build for an other PR.

Reply:
Force-pushed CI-cleaniness forgot to fmt the `use std::borrow::Borrow` behind feature gate